libraries/AC_AttitudeControl/AC_AttitudeControl.cpp

```diff
@@ -596,6 +580,7 @@ void AC_AttitudeControl::input_thrust_vector_rate_heading(const Vector3f& thrust
         Quaternion yaw_quat;
         yaw_quat.from_axis_angle(Vector3f{0.0f, 0.0f, heading_rate * _dt});
         _attitude_target = _attitude_target * thrust_vec_correction_quat * yaw_quat;
+        _attitude_target.to_euler(_euler_angle_target);
```

Comment:
Why are you adding this here?

Reply:
`_euler_angle_target` used in `ang_vel_to_euler_rate` on 591 before its updated in `attitude_controller_run_quat`

Reply:
In this case _ang_vel_target.zero();